### DISCUSSION TITLE
Radio migration made more easy.

### DISCUSSION THREAD

**Discussion by tech2mar** [Koenkk/zigbee2mqtt#17085] [General]
Hey friends,

I hope you are all doing well,
I'm searching for a method to migrate my config from my home server to a single rpi with z2m on it, and changing the radio at the same time from a sonoff p serial to some tubeszb on tcp.

I can't find a way to migrate the MAC Adress between both, I used the ti flasher but it seems not to work. Getting always a error on my pi.

Is it possible to restore a z2m config over the gui? Or to migrate the radios? I mean in the future. This would be a good thing.

**Comment by junosuarez** [comment]
You can get a backup from the web UI. This is a zip file with the contents of your data directory, which includes your yaml config and zigbee-herdsman database. With this, you should be and to copy the contents into the data directory in a new device, boot it up, and have everything "migrated". Since you mention changing the radio, you'll also have to make sure to update the config to point to the new adapter, if it's different. This will keep the network parameters like channel and key. 

A suggestion, If you separate it into two steps - migrating from the server to the raspberry pi with the existing radio, and then changing the radio - it will be easier to troubleshoot the cause of any errors that might arise.

**Reply by tech2mar** [reply]
I see, is there a plan for having a sort of restore on the UI? This would be great.
